README.md

```diff
@@ -1,69 +1,66 @@
 
-The [wasm3](https://github.com/wasm3/wasm3) runtime got chosen since it is written in C and has minimal dependencies (except a C library) and it is extremely portable. In kernel space, there is no libc, but we maintain a fork of wasm3 which can run in kernel space as well (check the `thrid-party/wasm3/` submodule).
+## Architecture
 
-Current restrictions for kernel-space wasm3:
-- no floating point support [can be soft-emulated if needed]
-- no WASI support
+Nasp's architecture consists of currently 2 different components: the kernel module and the agent. This will change in the future, we plan to add a control-plane, but the current architecture is the following:
 
-## Development environment
+![Nasp architecture](./docs/img/nasp-architecture.png)
 
-Our primary development environment is [Lima](https://lima-vm.io) since it supports x86_64 and ARM as well. Follow the instructions for [nasp-kernel-module](https://github.com/cisco-open/nasp-kernel-module) for setting up the development environment.
+## Components
 
-## CLI
+The Nasp kernel module comes with a user space [CLI](./cli/) written in Go. The kernel module exposes a character device: `/dev/nasp`, which is opened by the agent, the CLI talks with the agent. One usually runs this CLI on the Linux host itself.
 
-The Nasp kernel module comes with a user space [CLI](./cli/) written in Go. The kernel module exposes a character device: `/dev/wasm`, which can be used to interact with the module through the CLI. One usually runs this CLI on the Linux host itself.
+### Agent (server)
 
-### Build
+The agent is the server side of the CLI. It is responsible for the following:
+- communicates with the kernel module directly
+- parses and compiles the policy file to Wasm and loads to the kernel module
+- signs CSR requests from the kernel module
+- enhances process data with metadata from the host environment (e.g. Kubernetes, AWS, etc...)
+
+Usage:
 
 ```bash
-lima make build-cli
+sudo nasp agent
 ```
 
-## Examples
+### Client
 
-You can find a few sample applications compiled to Wasm and also EBPF code snippets which use them under the [samples](./samples) directory. All examples use the CLI underneath, inside the Make targets.
+```bash
+nasp load-policy --policy-file ./policy.json
+```
 
-### Simple Hello World
+## Development
 
-Build and load the Hello World Wasm module to the runtime inside the kernel module:
+### Development environment
 
-```bash
-make build-rust-wasm # This builds basically all Rust Wasm samples
-lima make load-hello-world-rust-wasm
-```
+Our primary development environment is [Lima](https://lima-vm.io) since it supports x86_64 and ARM as well. Follow the instructions for [nasp-kernel-module](https://github.com/cisco-open/nasp-kernel-module) for setting up the development environment.
```

Comment:
```suggestion
Our primary development environment is [Lima](https://lima-vm.io) since it supports x86_64 and ARM as well. Follow the instructions for [nasp-kernel-module](https://github.com/cisco-open/nasp-kernel-module#coding) for setting up the development environment.
```